scripts/host_modules/ztp_handler.py

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python3
+'''
+Copyright 2019 Broadcom. The term "Broadcom" refers to Broadcom Inc.
+and/or its subsidiaries.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import sys
+import argparse
+from argparse import RawTextHelpFormatter
+import os
+import subprocess
+import errno
+import time
+import datetime
+import signal
+from ztp.JsonReader import JsonReader
+from ztp.ZTPLib import getCfg, setCfg, getFeatures, getTimestamp
+from ztp.ZTPCfg import ZTPCfg
+import host_service
+import json
+import syslog 
+
+"""ZTP command handler"""
+
+MOD_NAME = 'ztp'
+
+class ZTP(host_service.HostModule):
+    """DBus endpoint that executes ZTP related commands
+    """
+    @staticmethod
+    def _run_command(commands):
+        """Run a ZTP command"""
+        cmd = ['/usr/bin/ztp']
+        if isinstance(commands, list):
+            cmd.extend(commands)
+        else:
+            cmd.append(commands)
+
+        try:
+            rc = 0
+            output = subprocess.check_output(cmd)
+        except subprocess.CalledProcessError as err:
+            rc = err.returncode
+            output = ""
+
+        return rc, output
+
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='', out_signature='')
+    def enable(self):
+        self._run_command("enable")
+
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='', out_signature='is')
+    def getcfg(self):
+        ztp_cfg=ZTPCfg()
+        ad = getCfg('admin-mode', ztp_cfg=ztp_cfg)
+        if ad == False:
+            return 0,"disabled"
+        else:
+            return 0,"enabled"
+
+
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='', out_signature='')
+    def disable(self):
+        self._run_command(["disable", "-y"])
+
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='', out_signature='is')
+    def status(self):
+        print("Calling ZTP Status")       
+        b = ztp_status()
+        print("Returning", b)
+        return 0,b
+        
+def register():
+    """Return the class name"""
+    return ZTP
+
+
+
+ztp_cfg = None
+## Signal handler is called on SIGTERM or SIGINT
+def signal_handler(signum, frame):
```

Comment:
Not sure we need this function - the top level host service should be the only entity listening on signals.